CUDADataFormats/BeamSpot/interface/BeamSpotCUDA.h

```diff
@@ -0,0 +1,32 @@
+#ifndef CUDADataFormats_BeamSpot_interface_BeamSpotCUDA_h
+#define CUDADataFormats_BeamSpot_interface_BeamSpotCUDA_h
+
+#include "HeterogeneousCore/CUDAUtilities/interface/device_unique_ptr.h"
+
+#include <cuda_runtime.h>
+
+class BeamSpotCUDA {
+public:
+  // alignas(128) doesn't really make sense as there is only one
+  // beamspot per event?
+  struct Data {
+    float x, y, z;  // position
+    // TODO: add covariance matrix
+
+    float sigmaZ;
+    float beamWidthX, beamWidthY;
+    float dxdz, dydz;
```

Comment:
related to https://github.com/cms-sw/cmssw/pull/29154#issuecomment-596692163
I think that it would be useful to have this struct in regular DataFormats/BeamSpot  as BeamSpotPOD or BeamSpotCore or something like that.
This way we could even make the "legacy" producer to write this, if it becomes useful. 

Related to the data members: should we expect a pattern that 3D vectors are going to be unrolled into individual data members?

Reply:
> Related to the data members: should we expect a pattern that 3D vectors are going to be unrolled into individual data members?

As you can see from the comment above, including the beamspot covariance matrix. I don't remember anymore why I did not include it in the end (I have vague recollection it would have had some relation to Eigen). Unrolling the covariance matrix is one option, but I suppose the exact technical implementation is open for suggestions.

> I think that it would be useful to have this struct in regular DataFormats/BeamSpot as BeamSpotPOD or BeamSpotCore or something like that.

Then, in principle, `cms::cuda::device::unique_ptr<BeamSpotPOD>` could be stored directly in the Event (*), and the `BeamSpotCUDA` class itself removed. I'm not saying we *should* do it though.

(*) ok, the product would really be `cms::cuda::Product<cms::cuda::device::unique_ptr<BeamSpotPOD>>`